src/uct/tcp/tcp_cm.c

```diff
@@ -527,7 +538,7 @@ uct_tcp_cm_handle_conn_req(uct_tcp_ep_t **ep_p,
 
 out:
     if (!(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_TX)) {
-        uct_tcp_ep_destroy_internal(&ep->super.super);
+        uct_tcp_ep_set_failed(ep);
```

Comment:
why not enough that the caller could check for *ep==NULL?

Reply:
fixed